File: Single/Easy/The.Descent.php
<?php
// The Descent
    $height = array();
    $maxHeight = 0;
    $canFire = true;
    while (TRUE) {
        fscanf(STDIN, "%d %d", $SX, $SY);
        $canFire = true;
        for ($i = 0;  $i < 8; $i++)
        {
            fscanf(STDIN, "%d", $MH);   // Represents height of one mountain from 9-0. Mountain heights provided left to right
            $height[$i] = $MH;
        }
        // Get max height
        for ($j = 0, $maxHeight = 0; $j < count($height); $j++) {
            if ($height[$j] > $maxHeight) {
                $maxHeight = $height[$j];
            }
        }
        if ($height[$SX] == $maxHeight && $canFire) {
            echo("FIRE\n");             // Either FIRE or HOLD
            $canFire = false;
        } else {
            echo("HOLD\n");         
        }
    }
?>

File: Single/Easy/The.Descent.java
// The Descent
import java.util.*;

class Player {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        // game loop
        int[] height = new int[8];
        int maxHeight = 0;
        boolean canFire = true;
        while (true) {
            int SX = in.nextInt();
            int SY = in.nextInt();
            canFire = true;
            for (int i = 0; i < 8; i++) {
                int MH = in.nextInt(); // represents the height of one mountain, from 9 to 0. Mountain heights are provided from left to right.
                height[i] = MH;
            }
            maxHeight = 0;  // get max height
            for (int j = 0; j < height.length; ++j) {
                if (height[j] > maxHeight) {
                    maxHeight = height[j];
                }
            }
            if (height[SX] == maxHeight && canFire) {
                System.out.println("FIRE"); // either: FIRE (ship is firing its phase cannons) or HOLD (ship is not firing).
                canFire = false;
            } else {
                System.out.println("HOLD");
            }
        }
    }
}